### PR TITLE
HPCC-3126 Rationalize ftslave and dafilesrv configuration 

### PR DESCRIPTION
**Pull request by g-pan**
Synapses of dafilesrv and ftslave, included in HPCC System Adminstrators guide. 

@jakesmith please review
@JamesDeFabia please review

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Completed Successful Build Test :  http://10.240.32.243/view/Docs%207.0/job/DocDev70/27/CE=ubuntu-14.04-amd64-1/

(HPCC Systems Administrators Guide)

### CODE REVIEW

File: docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  <sect3 id="DAFilesrv">
+    <title>dafilesrv</title>
+
+    <para>The dafilesrv is a daemon that runs on system servers or physical
+    locations on the HPCC system. This daemon enables the HPCC components to
+    have file access to physical file locations. File access could be
+    spraying, despraying, copying, etc. It is also used to enable file
+    security.</para>
+
+    <para>The dafilesrv daemon runs as a process on every Thor or Roxie node.
```
Comment (jakesmith) [inline docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml]: Not just Thor and Roxie nodes, all hpcc nodes may require it at some stage. Anything that requires a file from another node will go via dafilesrv - so e.g. hthor creates files access by others will require a dafilesrv on the eclagent nodes, esp accessing logs from any remote component will require a dafilesrv to be running on the node that/those components are running on.

File: docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    <para>You must have sudo permission level access in order to start or stop
+    any of the HPCC components, including dafilesrv.</para>
+  </sect3>
+
+  <sect3>
+    <title>ftslave</title>
+
+    <para>The ftslave is a support utility used by Thor slaves and eclagent
+    when deploying data to the nodes. The ftslave is launched on demand as
```
Comment (jakesmith) [inline docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml]: It is not used by Thor or eclagent - it is only a process used by spraying by Dfu server.

File: docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  tasks.</para>
+
+  <sect3 id="DAFilesrv">
+    <title>dafilesrv</title>
+
+    <para>The dafilesrv is a daemon that runs on system servers or physical
+    locations on the HPCC system. This daemon enables the HPCC components to
+    have file access to physical file locations. File access could be
+    spraying, despraying, copying, etc. It is also used to enable file
+    security.</para>
```
Comment (jakesmith) [inline docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml]: >It is also used to enable file security

That could be misleading. Or need to be clear that talking about physical vs logical security.
Dafilesrv can be configured to allow SSL connections only, which secures remote access to physical files.

**Comment by jakesmith** [comment]
@g-pan - a couple of comments.

**Comment by g-pan** [comment]
Excellent feedback @jakesmith I have incorporated it into the latest commit. 
@Michael-Gardner can you please review line # 31 for validity.

**Comment by g-pan** [comment]
@JamesDeFabia please review

File: docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml [hunk 4]
```diff
@@ -14,17 +14,19 @@
...
-    installation package. </para>
+    spraying, despraying, copying, accessing logs, etc. Anything that requires
+    a file from another node, must have dafilesrv running on that node. The
+    dafilesrv daemon can be configured to allow SSL connections only, which
+    secures remote access to physical files.</para>
+
+    <para>The dafilesrv daemon runs as a process on every HPCC node. The
+    daemon will typically keep running even if your system is stopped. That is
+    an important fact to keep in mind if you are stopping your system for
+    maintenance. Particularly for installations or upgrades. Even if you issue
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml]: Particularly for installations or upgrades. <-- sentence fragment

File: docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml [hunk 5]
```diff
@@ -14,17 +14,19 @@
...
+    the HPCC system stop command, you would still need to stop the dafilesrv
+    daemon. Issue the following command to stop the dafilesrv daemon. Then
+    verify that dafilesrv is not running before running the installation
+    package.</para>
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml]: 1) Doesn't the installer handle this automatically? 
2) The two sentences are better as a single sentence : Issue the following command to stop the dafilesrv daemon, then verify that dafilesrv is not running before running the installation package. 
3) Doesn't say HOW to verify that it is not running.

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
comments inline

**Comment by g-pan** [comment]
@Michael-Gardner after a few additional changes the line number changed. Could you please review line # 37 in the DAFtslave.xml file.

File: docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    <para>One way to check and see if dafilesrv is running, is to issue a
+    command such as the following:</para>
+
+    <para><programlisting>ps -eaf | grep dafilesrv</programlisting> Even if
+    your HPCC System is stopped, you should still see that the dafilesrv is
+    running. </para>
+
+    <para>Issue the following command to stop the dafilesrv daemon. </para>
+
+    <programlisting>sudo /etc/HPCCSystems/sbin/hpcc-run.sh -a dafilesrv stop</programlisting>
```
Comment (Michael-Gardner) [inline docs/EN_US/HPCCSystemAdmin/SA-Mods/DAFtslave.xml]: should probably also add the systemd alternative here?

Also should be OPT not ETC for the root of this.

**Comment by g-pan** [comment]
@Michael-Gardner please review latest revision.

**Review by JamesDeFabia — APPROVED** [review]
Looks reasonable from my POV. @Michael-Gardner should have the last word

**Comment by g-pan** [comment]
@Michael-Gardner could you please review the changes you requested.

**Comment by Michael-Gardner** [comment]
@g-pan Looks good to go!

**Pull request merged**